code/modules/surgery/bodyparts/bodyparts.dm

```diff
@@ -45,6 +45,21 @@
 		owner = null
 	return ..()
 
+/obj/item/bodypart/attack(mob/living/carbon/C, mob/user)
+	if(ishuman(C))
+		var/mob/living/carbon/human/H = C
+		if(H.dna && H.dna.species && EASYLIMBATTACHMENT in H.dna.species.specflags)
```

Comment:
just an fyi, you don't need to check for dna and dna.species, humans always have those. The only moment where a human doesn't have them is when he's being qdel'd.